coaster-nn/src/plugin.rs

```diff
@@ -144,13 +144,8 @@ pub trait ConvolutionContext<F> {}
 /// Provides Rnn Config functionality.
 ///
 /// Needs to be implemented for Operation specific configurations.
-pub trait RnnConfig<F> {
-    /// Workspace Size - Overwritten by each plugin method except native, which doesn't require
-    /// a workspace size.
-    fn workspace_size(&self) -> usize {
-        0
-    }
-}
+/// TODO: Rename to RnnContext since it contains mutable data.
+pub trait RnnConfig<F> {}
```

Comment:
We should be very diligent on requiring mutable configs. I don't think there is much prior art on mutable execution contexts for layers.